Matter_Engine/Query/GeneralMaterialTakeoff.cs

```diff
@@ -0,0 +1,110 @@
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using BH.oM.Physical.Materials;
+using BH.oM.Quantities.Attributes;
+using BH.oM.Base.Attributes;
+using BH.Engine.Base;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.Engine.Matter
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Gets the unique Materials along with their volumes defining an object's make-up.")]
+        [Input("elementM", "The element to get the GeneralMaterialTakeoff from.")]
+        [Input("checkForTakeoffFragment", "If true and the provided element is a BHoMObject, the incoming item is checked if it has a VolumetricMaterialTakeoff fragment attached, and if so, returns a GeneralMaterialTakeoff based on it. If false, the GeneralMaterialTakeoff returned will be calculated, independant of fragment attached.")]
+        [Output("generalMaterialTakeoff", "The kind of matter the element is composed of and in which quantities.")]
+        public static GeneralMaterialTakeoff IGeneralMaterialTakeoff(this IElementM elementM, bool checkForTakeoffFragment = true)
+        {
+            if (elementM == null)
+            {
+                Base.Compute.RecordError("Cannot query the VolumetricMaterialTakeoff from a null element.");
+                return null;
+            }
+
+            //If asked to look for fragment, and if fragment exists, return it
+
+            if (checkForTakeoffFragment)
+            {
+                VolumetricMaterialTakeoff volMatTakeoff;
+                if (TryGetVolumetricMaterialTakeoffFragment(elementM, out volMatTakeoff))
+                    return Create.GeneralMaterialTakeoff(volMatTakeoff);
+            }
+            //IElementMs should implement one of the following:
+            // -SolidVolume and MaterialComposition or
+            // -VolumetricMaterialTakeoff
+            //- GeneralMaterialTakeoff
+            //This method first checks if the VolumetricMaterialTakeoff method can be found and run, and if so uses it.
+            //If not, it falls back to running the MaterialComposition and SolidVolume methods and gets the VolumetricMaterialTakeoff from them.
```

Comment:
@IsakNaslundBh worth adding this to the documentation for required extension methods post-merge

Reply:
Yes, was planning to, but good reminder. Thanks :)